mainapp/views.py

```diff
@@ -181,7 +182,8 @@ def mapdata(request):
     if district != "all":
         data = Request.objects.exclude(latlng__exact="").filter(district=district).values()
     else:
-        data = Request.objects.exclude(latlng__exact="").values()
+        from_date = datetime.today() - timedelta(hours=24)
```

Comment:
`date.today() Return the current local date.` Server time can be in another time zone.

Reply:
you can use `from_date = datetime.now(pytz.timezone('Asia/Kolkata')) - timedelta(hours=24)`

Reply:
I haven't gone through the codebase, but this will only matter if the `dateadded` column was timezone aware in the first place.

Reply:
```
from django.utils import timezone
today = timezone.now().date()
```
Always use this

Reply:
Assume server timezone is utc pls. This code is correct afaik.

Reply:
is there a time.now.utc?

Reply:
```
from django.utils import timezone
today = timezone.now().date()
```
Always use timezone

Reply:
```
import datetime
datetime.timedelta(hours=4)
```
Import datetime module and use it like datetime.timedelta instead of directly importing it.

Reply:
Btw, `dateadded` is a `DateTimeField.` So you don't need to convert it to date first. Just do the following.
```
from_time = timezone.now() - datetime.timedelta(hours=24)
Request.objects.filter(dateadded__gt=from_time).exclude(latlng__exact="").values()
```
Otherwise you won't get correct results.

Reply:
Timezone it set in settings.py to ours. Check

Reply:
From the last prod db dump:
```Indexes:
    "mainapp_request_pkey" PRIMARY KEY, btree (id)
    "mainapp_request_district_idx" btree (district)
    "mainapp_request_requestee_phone_idx" btree (requestee_phone)```

We'll need indexes on `dateadded` and `latlng`

Reply:
I assume server, DB etc are configured to use UTC, hence this should work

Reply:
Either way the intention is not to filter for exact 24 hrs, but rather to reduce the no of records returned.